concent_api/conductor/tests/test_integration_verification.py

```diff
@@ -0,0 +1,276 @@
+
+        self.assertEqual(UploadRequest.objects.count(),   2)
+
+        for file in files:
+            self.assertTrue(UploadRequest.objects.filter(path = file).exists())
+
+    def test_verification_request_task_should_link_upload_requests_to_related_upload_reports(self):
+        files = [
+            'blender/result/ef0dc1/ef0dc1.zzz523.zip',
+        ]
+
+        upload_report = UploadReport(
+            path = files[0],
+        )
+        upload_report.full_clean()
+        upload_report.save()
+
+        # Create second UploadReport so view doesn't call verification_order_task.
+        upload_report_second = UploadReport(
+            path = 'blender/result/ef0dc1/ef0dc1.aaa523.zip',
+        )
+        upload_report_second.full_clean()
+        upload_report_second.save()
+
+        verification_request_task(
+            compute_task_def = self.compute_task_def,
+            files            = files,
+        )
+
+        self.assertEqual(VerificationRequest.objects.count(), 1)
+
+        verification_request = VerificationRequest.objects.first()
+        self.assertEqual(verification_request.subtask_id,               self.compute_task_def['subtask_id'])
+        self.assertEqual(verification_request.upload_requests.count(),  1)
+
+        self.assertEqual(UploadRequest.objects.count(), 1)
+
+        for file in files:
+            self.assertTrue(UploadRequest.objects.filter(path = file).exists())
+
+            upload_request = UploadRequest.objects.filter(path = file).first()
+            self.assertEqual(upload_request.upload_reports.count(), 1)
+
+    def test_verification_request_task_should_not_link_upload_requests_to_unrelated_upload_reports(self):
+        files = [
+            'blender/result/ef0dc1/ef0dc1.zzz523.zip',
+        ]
+
+        upload_report = UploadReport(
+            path='blender/result/ef0dc1/ef0dc1.aaa523.zip',
+        )
+        upload_report.full_clean()
+        upload_report.save()
+
+        verification_request_task(
+            compute_task_def = self.compute_task_def,
+            files            = files,
+        )
+
+        self.assertEqual(VerificationRequest.objects.count(), 1)
+
+        verification_request = VerificationRequest.objects.first()
+        self.assertEqual(verification_request.subtask_id,               self.compute_task_def['subtask_id'])
+        self.assertEqual(verification_request.upload_requests.count(),  1)
+
+        self.assertEqual(UploadRequest.objects.count(), 1)
+
+        for file in files:
+            self.assertTrue(UploadRequest.objects.filter(path = file).exists())
+
+            upload_request = UploadRequest.objects.filter(path = file).first()
+            self.assertEqual(upload_request.upload_reports.count(), 0)
+
+    @override_settings(CELERY_ALWAYS_EAGER = True)
+    def test_verification_request_task_should_schedule_verification_order_task_if_all_related_upload_requests_have_reports(self):
```

Comment:
how is `verification_order_task` scheduled? few lines below there is:

>  #Create second UploadReport so view doesn't call verification_order_task.
        upload_request_second = UploadReport(
            path = 'blender/result/ef0dc1/ef0dc1.aaa523.zip',
        )

I am afraid I do not understand this test?

Reply:
This was not needed and skipped because of bug in logic, now it is removed.